### ISSUE TITLE
composer issue identity needs to be able to provide affiliation

### ISSUE THREAD

**Issue by davidkel** [hyperledger/composer#606]
reported in stackoverflow
https://stackoverflow.com/questions/43145965/cant-issue-identities-on-fabric-bluemix-instance-for-fabric-composer-business-n

Also this will be an issue when moving to Hyperledger fabric V1 as fabric-ca-server will not allow you to create identities without providing an affiliation defined by the server.

**Comment by sstone1** [comment]
We have to be mindful when fixing this that we do not add Fabric specific options to the `composer identity issue` command, nor the underlying APIs and playground panels (all of which may need fixing). Looking at the latest v1.0 code, we need to support `affiliation`, `maxEnrollments` and `role` which are all Fabric specific options.

The easiest thing to do may be to expose the default values for these options via additional connection profile options and see if/how users want to supply the values via the CLI.

**Comment by davidkel** [comment]
The chances are there will be services out there implementing a fabric-ca that won't use the dummy affiliations defined and unless we have some way of specifying the affiliation then identity issue is not going to work. So I think this is a P1.

**Comment by davidkel** [comment]
Playground is going to need to same capability

**Comment by sstone1** [comment]
The work required here is similar to the work required in:
As a network starter, I can specify a HLF v1.0 endorsement policy during deployment #1251

Affiliation is a Hyperledger Fabric specific concept and does not belong in the Composer APIs or CLIs. Instead, we need the capability to specify connector specific options as part of the `composer identity issue` command.

`BusinessNetworkConnection.issueIdentity(participantId,newUserId[,options])` already accepts an optional set of connector specific options - so most of the work is already done.

The CLI then needs to be updated with two optional arguments that allow these options to be specified via the command line or via a configuration file:
composer identity issue -p hlfv1 -n biznet -u maeid1 -a participant1 -o connectorOption1=someValue -o connectorOption2=someOtherValue
composer identity issue -p hlfv1 -n biznet -u maeid1 -a participant1 -O connectorOptionFile.txt

In the case of the first example, the options argument to BusinessNetworkConnection.issueIdentity would look as follows:

```
{
  "connectorOption1":"someValue",
  "connectorOption2":"someOtherValue"
}
```

The HLFv1 connector can then be updated to look for `affiliation=someOrg` options and include them in the issue identity request.

**Comment by davidkel** [comment]
possible options for hlfv1 are
affiliation=
role=
maxEnrollments
attributes=

affiliation=example.com
maxEnrollments=10
role=admin
attributes='{"attr1":"value1", "attr2":123}'

**Comment by davidkel** [comment]
Now available in 0.9.1
